upgrades/1.3-1.4/common/migrate_product_template.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+use Pim\Upgrade\SchemaHelper;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Output\ConsoleOutput;
+
+require_once __DIR__.'/../../app/bootstrap.php.cache';
+require_once __DIR__.'/../../app/AppKernel.php';
+
+class MigrationProductTemplate
+{
+    protected $output;
+    protected $env;
+    protected $container;
+    protected $kernel;
+    protected $productTemplateTable;
+
+    public function __construct(ConsoleOutput $output, ArgvInput $input)
+    {
+        $this->output = $output;
+
+        $env = $input->getParameterOption(['-e', '--env']);
+        if (!$env) {
+            $env = 'dev';
+        }
+
+        $this->kernel($env);
+
+        $schemaHelper = new SchemaHelper($this->container);
+        $this->productTemplateTable = $schemaHelper->getTableOrCollection('product_template');
+    }
+
+    public function execute()
+    {
+        $productTemplates = $this->getProductTemplates();
+        if (empty($productTemplates)) {
+            $this->output->writeln('<info>There is no product template to update<info>');
+
+            return;
+        }
+
+        $this->convert($productTemplates);
+        $this->output->writeln('<info>Done !</info>');
+    }
+
+    /**
+     * @param array $productTemplates
+     *
+     * @return array
+     */
+    protected function convert(array $productTemplates)
+    {
+        $connection = $this->getConnection();
+
+        foreach ($productTemplates as $template) {
+            $data = [];
+            foreach ($template->getValuesData() as $code => $values) {
+                foreach ($values as $i => $value) {
+                    if (array_key_exists('value', $value)) {
+                        $data[$code][$i] = [
+                            'scope'  => $value['scope'],
+                            'locale' => $value['locale'],
+                            'data'   => $value['value']
+                        ];
+                    }
+                }
+            }
+
+            if (!empty($data)) {
+                $sql = sprintf("UPDATE %s SET valuesData=:data WHERE id = :id", $this->productTemplateTable);
+
+                $stmt = $connection->prepare($sql);
+                $stmt->bindValue('data', json_encode($data));
+                $stmt->bindValue('id', $template->getId());
+                $stmt->execute();
+            }
+        }
+    }
+
+    /**
+     * @return array
+     */
+    protected function getProductTemplates()
+    {
+        return $this->container->get('doctrine.orm.default_entity_manager')
+            ->getRepository($this->container->getParameter('pim_catalog.entity.product_template.class'))
+            ->findAll();
+    }
+
+    /**
+     * Load kernel
+     *
+     * @param string $env
+     */
+    public function kernel($env = 'dev')
```

Comment:
First I was "wow, a function with the exact same name as a property, how does this work". As this function is only used in the constructor I would prefer to see this code inside the constructor instead of a function that can be called everywhere in the class.